ovos_dinkum_listener/service.py

```diff
@@ -726,5 +760,28 @@ def _handle_opm_vad_query(self, message):
         }
         self.bus.emit(message.response(data))
 
-
+    def reload_configuration(self):
+        """
+        Reload configuration and restart loop. Automatically called when
+        Configuration object reports a change
+        """
+        with self._load_lock:
+            if self._config_hash() != self._applied_config_hash:
+                LOG.info(f"Listener configuration changed. Reloading")
+            else:
+                LOG.debug("Configuration not changed")
+                return
+
+            # Configuration changed, update status and reload
+            self.status.set_alive()
+            # Reload STT
+            if hasattr(self.stt, "shutdown"):
+                self.stt.shutdown()
+            self.stt = load_stt_module()
```

Comment:
can we do this per section? only reload the components that are actually needed

will need hashes per section
- stt
- fallback stt
- hotwords
- vad / microphone / listener

Reply:
Maybe.. There are also global config params that should be watched, like `lang` that would affect all of these

Reply:
Updated and I set hotwords to change on any listener config changes to account for default ww/wakeup, upload settings, and anything else I'm forgetting